src/errors.rs

```diff
@@ -47,6 +50,145 @@ impl Display for ErrorType {
     }
 }
 
+/// Custom result class, that supports extended errors.
+/// Returning Error variants will result in throwing errors in JS.
+///
+/// You can either use Converted error, that supports some extra fields,
+/// or you can add a new error type. Note that `with_custom_error!` macro
+/// supports only ConvertedError conversion. Usage with different error requires manual handling.
+/// This is important, because we cannot implement ToNapiValue for Result<T, E> directly,
+/// And we cannot implement FromResidual for this class, as this is unstable feature.
+///
+/// When converting this class to JS value through ToNapiValue
+/// cases will be handled in the following way
+/// - Ok(T): T::to_napi_value will be return
+/// - Error(ConvertedError): We will create napi::Error based on ConvertedError ToNapiValue trait and return Err(Error)
+/// - NapiError(Error): Err(Error) will be returned
+pub enum JsResult<T> {
+    Ok(T),
+    Error(ConvertedError),
```

Comment:
:thought_balloon: Shouldn't this rather be `ConvertibleError`? It's not converted at the point of construction, right?

Reply:
OK, I think `ConvertedError` is a valid name.